### ISSUE TITLE
Failure to build jsonff

### ISSUE THREAD

**Issue by everythingfunctional** [fortran-lang/fpm#265]
I wanted to see how well the Fortran version was doing (I haven't really used it at all yet). I tried building my [jsonff](https://gitlab.com/everythingfunctional/jsonff) package and got the following error.

```
 + gfortran -c build/dependencies/iso_varying_string/src/ISO_VARYING_STRING.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_iso_varying_string_src_ISO_VARYING_STRING.f90.o
 + gfortran -c build/dependencies/erloff/src/Module_m.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_erloff_src_Module_m.f90.o
 + gfortran -c build/dependencies/erloff/src/Procedure_m.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_erloff_src_Procedure_m.f90.o
 + gfortran -c build/dependencies/strff/src/strff.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_strff_src_strff.f90.o
 + gfortran -c build/dependencies/erloff/src/Call_stack_entry_m.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_erloff_src_Call_stack_entry_m.f90.o
 + gfortran -c build/dependencies/erloff/src/Call_stack_m.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_erloff_src_Call_stack_m.f90.o
 + gfortran -c build/dependencies/erloff/src/Message_m.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_erloff_src_Message_m.f90.o
 + gfortran -c build/dependencies/erloff/src/Error_list_m.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_erloff_src_Error_list_m.f90.o
 + gfortran -c build/dependencies/erloff/src/Message_list_m.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_erloff_src_Message_list_m.f90.o
 + gfortran -c build/dependencies/erloff/src/erloff.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_erloff_src_erloff.f90.o
 + gfortran -c build/dependencies/parff/src/parff.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/dependencies_parff_src_parff.f90.o
 + gfortran -c ./src/jsonff.f90 -Wall -Wextra -Wimplicit-interface  -fPIC -fmax-errors=1 -g -fbounds-check -fcheck-array-temporaries -fbacktrace -Jbuild/gfortran_debug/jsonff -o build/gfortran_debug/jsonff/jsonff.f90.o
./src/jsonff.f90:46:17:

   46 |             join, &
      |                 1
Error: Symbol 'readfile' referenced at (1) not found in module 'strff'
compilation terminated due to -fmax-errors=1.
 Command failed
ERROR STOP 

Error termination. Backtrace:
#0  0x101fc4ebd
#1  0x101fc5b75
#2  0x101fc6f03
#3  0x101f109f8
#4  0x101efd552
#5  0x101efcc32
#6  0x101efdffd
#7  0x101ef8ddd
#8  0x101f92886
```

This error is from MacOS, but a similar one is emitted on Linux. It appears it is fetching the incorrect version of `strff`. It fetched version 1.0.0, but version 1.2.0 is specified in the `fpm.toml`. Maybe it's fetching dependencies depth first instead of breadth first?

**Comment by LKedward** [comment]
Thanks for reporting, yes it's fetching depth first. Since we don't (yet) check for conflicting versions in dependencies, is breadth first the correct/preferred way of fetching?

**Comment by everythingfunctional** [comment]
By doing breadth first, the user can manually specify a specific version for a dependency if there are any conflicts, because it fetches exactly what they've specified first.

**Comment by LKedward** [comment]
Ah I see, thanks for explaining.

**Comment by awvwgk** [comment]
I was using `jsonff` to test my dependency resolution, so this issue will probably be fixed with #266.
